### PR TITLE
Merge Une-T INS improvements - remove hardcoded values that should be in variables

### PR DESCRIPTION
**Pull request by franck-boullier**
This is needed so that the codebase for Unee-T is easier to deploy on different installation.

We are importing some of the improvements made in the Unee-T INS fork.
We are also making sure that we are using variables instead of hardcoded values as much as possible.

# Pre-requisite:

## AWS variables in the parameter store (DONE):

We MUST have the correct values for the following variables in the AWS parameter store for each environment: DEV (DONE), PROD (DONE), DEMO (DONE)
- DOMAIN
- INSTALLATION_ID
- STAGE
- EMAIL_FOR_NOTIFICATION_UNIT
- PRIVATE_SUBNET_1
- PRIVATE_SUBNET_2
- PRIVATE_SUBNET_3
- LAMBDA_TO_RDS_SECURITY_GROUP
- API_ACCESS_TOKEN

## Travis CI Setting (DONE):
We MUST have the following settings in Travis CI for this repo.
- AWS_DEFAULT_REGION
- GITHUB_TOKEN
- AWS_ACCOUNT_USER_ID_DEV
- AWS_ACCOUNT_SECRET_DEV
- AWS_PROFILE_DEV
- AWS_ACCOUNT_USER_ID_DEMO
- AWS_ACCOUNT_SECRET_DEMO
- AWS_PROFILE_DEMO
- AWS_ACCOUNT_USER_ID_PROD
- AWS_ACCOUNT_SECRET_PROD
- AWS_PROFILE_PROD

### CODE REVIEW

**Comment by franck-boullier** [comment]
@kaihendry I have a weird issue with Travis CI that doesn't pass the test for reasons I can't understand.

See the [Travis CI build](https://travis-ci.org/unee-t/unit/builds/644956559#L456) for more details.

What's weird is that sometimes things go as expcted and sometime I have error messages for seamingly random reasons...
Here the first test PASS but the other 2 FAIL
image

Here all 3 test PASS
image

Any idea? Could this be an issue with the latest version of uneet/env where the connection parameters needed to access the database seem to be defined (see the function `BugzillaDSN`?

I'd love your help as I'm stuck here...

**Comment by franck-boullier** [comment]
Travis CI fails most likely because of timeout issue <-- restarting the same build after a short while usually works!

File: .travis.yml [hunk 1]
```diff
@@ -1,50 +1,67 @@
...
-  - go test -v ./...
+ # We configure a specific AWS profile for testing
+ # This will ONLY test the DEV configuration at this stage!
+ - aws configure set profile.travis-test-dev.aws_access_key_id $AWS_ACCOUNT_USER_ID_DEV
+ - aws configure set profile.travis-test-dev.aws_secret_access_key $AWS_ACCOUNT_SECRET_DEV
+ - aws configure set profile.travis-test-dev.region $AWS_DEFAULT_REGION
+ - aws configure --profile travis-test-dev list
+ # We run the script that will define the environment variables we need for the test
+ - export TRAVIS_AWS_PROFILE=travis-test-dev
+ - ./aws.env
```
Comment (kaihendry) [inline .travis.yml]: How about a test after here to check the env is actually exported?

Comment (franck-boullier) [reply .travis.yml]: This is a good idea, I have one already in theory in the `aws.env` script itself but we can probably get something more robust...
Not sure how to do that yet, I'll have to research that.

File: main.go [hunk 2]
```diff
@@ -77,29 +77,49 @@ func (h handler) step1Insert(unit unit) (err error) {
 	return
 }
 
-// New setups the configuration assuming various parameters have been setup in the AWS account
-func New() (h handler, err error) {
+// NewDbConnexion setups the configuration assuming various parameters have been setup in the AWS account
+// TO DO: REVIEW THIS A LOT OF HARDCODED VALUES HERE...
+func NewDbConnexion() (h handler, err error) {
 
 	cfg, err := external.LoadDefaultAWSConfig(external.WithSharedConfigProfile("uneet-dev"))
```
Comment (franck-boullier) [inline main.go]: @kaihendry I'm almost done with this PR.
One thing I don't understand is why you need these here? What is the objective of doing that?

**Comment by franck-boullier** [comment]
@kaihendry I'm done with this PR for now and I'm ready to merge.

Do you see any reason why I should NOT merge this at this stage?

**Pull request merged**